clients/spark/core/src/test/resources/parser-test/sst_files_generator.go

```diff
@@ -0,0 +1,291 @@
+
+	// TableFormat specifies the format version for writing sstables. The default
+	// is TableFormatRocksDBv2 which creates RocksDB compatible sstables. this is the only format supported by lakeFS.
+	writerOptions = sstable.WriterOptions{
+		Compression: sstable.SnappyCompression,
+		TableFormat: sstable.TableFormatLevelDB,
+	}
+	createTestInputFiles(keys, generateNanoid, sizeBytes,
+		"table.format.leveldb", writerOptions)
+}
+
+func writeSstsWithWriterOptionsFuzzing() {
+	sizeBytes := DefaultSstSizeBytes
+	keys := prepareSortedSliceWithNanoid(sizeBytes)
+
+	// Use fuzzing to define sstable user properties.
+	generateFuzz := newGenerateFuzz()
+	userProps := map[string]string{}
+	const numOfUserDefinedProps = 20
+	for i := 1; i <= numOfUserDefinedProps; i++ {
+		propKey, _ := generateFuzz()
+		propVal, _ := generateFuzz()
+		userProps[propKey] = propVal
+		fmt.Printf("user property %d, key=%s, val=%s\n", i, propKey, propVal)
+	}
+	writerOptions := sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: []func() sstable.TablePropertyCollector{NewStaticCollector(userProps)},
+	}
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "fuzz.table.properties", writerOptions)
+
+	// BlockRestartInterval is the number of keys between restart points
+	// for delta encoding of keys.
+	// The default value is 16.
+	blockRestartInterval := 1 + rand.Intn(100)
+	writerOptions = sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: []func() sstable.TablePropertyCollector{NewStaticCollector(DefaultUserProperties)},
+		BlockRestartInterval:    blockRestartInterval,
+	}
+	fmt.Printf("BlockRestartInterval %d...\n", blockRestartInterval)
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "fuzz.block.restart.interval", writerOptions)
+
+	// BlockSize is the target uncompressed size in bytes of each table block.
+	// The default value is 4096.
+	blockSize := 1 + rand.Intn(9000)
+	writerOptions = sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: []func() sstable.TablePropertyCollector{NewStaticCollector(DefaultUserProperties)},
+		BlockSize:               blockSize,
+	}
+	fmt.Printf("BlockSize %d...\n", blockSize)
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "fuzz.block.size", writerOptions)
+
+	// BlockSizeThreshold finishes a block if the block size is larger than the
+	// specified percentage of the target block size and adding the next entry
+	// would cause the block to be larger than the target block size.
+	//
+	// The default value is 90
+	blockSizeThreshold := 1 + rand.Intn(100)
+	writerOptions = sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: getDefaultTablePropertyCollectors(DefaultUserProperties),
+		BlockSizeThreshold:      blockSizeThreshold,
+	}
+	fmt.Printf("BlockSizeThreshold %d...\n", blockSizeThreshold)
```

Comment:
I believe that it may make more sense to mix all of the WriterOptions together.  So create X different files, with random values for each.